### PR TITLE
Docs firebase rework

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Transition Gloo docs to firebase hosting

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3834

### CODE REVIEW

File: docs/build-docs.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+###################################################################################
+# This script generates a versioned docs website for Gloo Edge which can 
+# then be deployed to firebase
+###################################################################################
+
+set -ex
+
+# Update this array with all versions of SMH to include in the versioned docs website.
```
Comment (kdorosh) [inline docs/build-docs.sh:10]: still references SMH

File: docs/old-version-marker.sh [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-
-# Find all html files and look for the <div class=burger> line that is part of the standard header
-# and add the warning message
-find . -name "*.html" | tr '\n' '\0' | xargs -0 -n 1 sed -i '' -e 's#<div class="burger">#<div><mark>You are on a previous version of Gloo. The most current docs can be found <a href="https://docs.solo.io/gloo/latest/">here</a>.</mark></div><div class="burger">#'
-
-# Add the robots.txt file to the root of the web site
-cat <<EOF >> robots.txt
-User-agent: *
-Disallow: /
-EOF
```
Comment (kdorosh) [inline docs/old-version-marker.sh:18]: we don't want to lose robots.txt on the older versions of the docs site

File: docs/conf/conf.d/default.conf [hunk 3]
```diff
@@ -1,36 +0,0 @@
...
-
-server {
-  listen 8080;
-
-  gzip on;
-  gzip_types text/plain text/css text/html application/javascript application/json;
-  gzip_comp_level 6;
-  gzip_buffers 16 8k;
-  gzip_vary on;
-  gzip_min_length 1100;
```
Comment (kdorosh) [inline docs/conf/conf.d/default.conf:15]: we don't want to lose gzip, this gave us a nice performance improvement

**Review by kdorosh — CHANGES_REQUESTED** [review]
some small things we still want to port over, looks good so far!

File: docs/conf/conf.d/default.conf (hunk 3 above)
Comment (EItanya) [reply docs/conf/conf.d/default.conf:15]: This is nginx config, no longer valid

File: docs/old-version-marker.sh (hunk 2 above)
Comment (EItanya) [reply docs/old-version-marker.sh:18]: I could not find how this was used anywhere. Can you show me how it's being used currently in old versions so I can make sure it stays in

Comment (kdorosh) [reply docs/old-version-marker.sh:18]: we were initially misusing it, we probably want to use https://developers.google.com/search/reference/robots_meta_tag instead

File: docs/conf/conf.d/default.conf (hunk 3 above)
Comment (kdorosh) [reply docs/conf/conf.d/default.conf:15]: by default, firebase gzips your files https://stackoverflow.com/questions/34318491/serving-gzipped-files-on-firebase-hosting so happy to resolve this

File: docs/old-version-marker.sh (hunk 2 above)
Comment (EItanya) [reply docs/old-version-marker.sh:18]: This will require overwriting another layout most likely, is this something we could potentially do in a follow-up?

Comment (kdorosh) [reply docs/old-version-marker.sh:18]: sure, opened https://github.com/solo-io/gloo/issues/3847 to track

**Review by kdorosh — APPROVED** [review]

**Pull request merged**